src/App.jsx

```diff
@@ -14,33 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const LENGTH = 'length';
+const ALPHABET = 'alphabet';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getPreparedGoods = (goods, { sortedField, reversed }) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortedField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortedField) {
+        case ALPHABET:
+          return good1.localeCompare(good2);
+        case LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortedField, setSortedField] = useState(false);
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortedField,
+    reversed,
+  });
+
+  const isResetVisible = sortedField !== false || reversed !== false;
+
+  const resetState = () => {
+    setSortedField(false);
+    setReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortedField(ALPHABET)}
+          type="button"
+          className={cn(
+            { 'is-light': sortedField !== ALPHABET },
+            'button is-info',
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortedField(LENGTH)}
+          type="button"
+          className={cn(
+            { 'is-light': sortedField !== LENGTH },
+            'button is-success',
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReversed(value => !value)}
+          type="button"
+          className={cn({ 'is-light': !reversed }, 'button is-warning')}
```

Comment:
usually more understandable and readable when static classes that are always applied are placed before conditional classes.

ex. : 
 className={cn('button is-warning', { 'is-light': !reversed })}